Automattic/jetpack
# CRM: Update register_step to not need slug parameter

## Proposed changes:
* This PR removes the `$step_name` parameter from `register_step` method in `Automattic\Jetpack\CRM\Automation\Automation_Engine`.

### Other information:


## Jetpack product discussion
https://github.com/Automattic/zero-bs-crm/issues/3270

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
* Open the class `Automattic\Jetpack\CRM\Automation\Automation_Engine`
* Make sure the method `register_step` no longer has the `$step_name` parameter 
* Make sure I didn't break anything in the process
